common/lib/xmodule/xmodule/course_module.py

```diff
@@ -910,6 +918,48 @@ def __init__(self, *args, **kwargs):
             self.discussion_topics = {_('General'): {'id': self.location.html_id()}}
 
         try:
+            enrollment_track_scheme = UserPartition.get_scheme("enrollment_track")
+            enrollment_track_partitions = self.get_user_partitions_for_scheme(enrollment_track_scheme)
+            # Note that when a course_module is initially created, the ID is the draft version.
+            # We just want to store the course_id without published or draft.
+            course_id = unicode(self.id.for_branch(None))
+            create_partition = False
+            new_id = 0
+
+            if len(enrollment_track_partitions) == 0:
+                used_ids = set(p.id for p in self.user_partitions)
+                while new_id in used_ids:
+                    new_id += 1
+                create_partition = True
```

Comment:
We talked about this in person, but recording here too. I don't think there is a race condition issue because each course_module instance that is created will persist its own version of user_partitions. It is possible (and quite likely in production) that multiple ones will add a enrollment_track partition to their user_partitions variable, but they will all be equivalent enrollment_track partitions, and which is the last one saved will just overwrite all the others. There will be no merging of user_partitions variables across multiple instances of course_module.